flowmachine/flowmachine/features/subscriber/active_subscribers.py

```diff
@@ -0,0 +1,120 @@
+import datetime
+from datetime import timedelta, date, datetime
+from typing import List, Union, Optional
+from flowmachine.core.query import Query
+from flowmachine.features.subscriber.call_days import CallDays
+from flowmachine.features.subscriber.interevent_interval import IntereventInterval
+from flowmachine.features.utilities.events_tables_union import EventsTablesUnion
+
+
+class ActiveSubscribers(Query):
+    """Returns a list of subscribers active `active_days`
+    out of `interval`, optionally with at least Z call-hours active"""
+
+    # TODO: Parameterise which events tables to use + which ID method to use
+
+    def __init__(
+        self,
+        start_date: Union[date, str],
+        end_date: Union[date, str],
+        active_days: int,
+        interval: int,
+        active_hours: int = None,
+        subscriber_id: str = "msisdn",
+        events_tables: Optional[List[str]] = None,
+    ):
+        self.start_date = start_date
+        self.end_date = end_date
+        self.active_days = active_days
+        self.interval = interval
+        self.active_hours = active_hours
+        self.sub_id_column = subscriber_id
+        self._window_start = self._start_date - timedelta(days=self.interval - 1)
+        self.events_table = EventsTablesUnion(
+            start_date,
+            end_date,
+            tables=events_tables,
+            subscriber_identifier=subscriber_id,
+            columns=[subscriber_id, "datetime"],
+        )
+        super().__init__()
+
+    @property
+    def start_date(self):
```

Comment:
Use `standardise_date` func on the input  instead of this?

Reply:
Will do